package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh

```diff
@@ -42,10 +42,17 @@ proto_gluon_bat0_setup() {
 	echo 1 > /proc/sys/net/ipv6/conf/primary0/disable_ipv6
 	ip link set primary0 address "$primary0_mac" mtu 1532 up
 
-	local routing_algo="$(uci -q get batman-adv.bat0.routing_algo || echo 'BATMAN_IV')"
-	(echo "$routing_algo" >/sys/module/batman_adv/parameters/routing_algo) 2>/dev/null
-
-	echo bat0 > /sys/class/net/primary0/batman_adv/mesh_iface
+	local routing_algo="$(uci -q get batman-adv.bat0.routing_algo || uci -q get batman-adv-legacy.bat0.routing_algo || echo 'BATMAN_IV')"
```

Comment:
You need to read from the site config whether batman-adv or batman-adv-legacy is to be used (and I don't think it makes sense to ever store the routing algo in /etc/config/batman-adv-legacy at all).